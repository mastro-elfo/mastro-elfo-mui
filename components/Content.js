"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Content;

var _react = _interopRequireDefault(require("react"));

var _core = require("@material-ui/core");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

/**
 * Puts a `Box` inside a `Container`.
 * @param       {[type]} children            [description]
 * @param       {Object} [ContainerProps={}] [description]
 * @param       {Object} [BoxProps={}}]      [description]
 * @constructor
 */
function Content(_ref) {
  var children = _ref.children,
      _ref$ContainerProps = _ref.ContainerProps,
      ContainerProps = _ref$ContainerProps === void 0 ? {} : _ref$ContainerProps,
      _ref$BoxProps = _ref.BoxProps,
      BoxProps = _ref$BoxProps === void 0 ? {} : _ref$BoxProps;
  return /*#__PURE__*/_react["default"].createElement(_core.Container, ContainerProps, /*#__PURE__*/_react["default"].createElement(_core.Box, _extends({
    pt: 1
  }, BoxProps), children));
}