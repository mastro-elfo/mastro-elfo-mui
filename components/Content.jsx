/**
 * Puts a `Box` inside a `Container`.
 *
 * Children are rendered inside box and all other props are passed to `Box` element.
 *
 * @see: https://material-ui.com/api/box/
 * @see: https://material-ui.com/api/container/
 */

import React from "react";

import { Box, Container } from "@material-ui/core";

export default function Content({
  children,
  ContainerProps = {},
  BoxProps = {}
}) {
  return (
    <Container {...ContainerProps}>
      <Box pt={1} {...BoxProps}>
        {children}
      </Box>
    </Container>
  );
}
