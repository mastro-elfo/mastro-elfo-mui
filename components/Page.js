"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Page;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ = require("./");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * Create a single page with header, content and a "printable" version (hidden on screen).
 *
 * By default scrolling the page make a `TopFab` appear.
 */
function Page(_ref) {
  var _ref$content = _ref.content,
      content = _ref$content === void 0 ? null : _ref$content,
      _ref$header = _ref.header,
      header = _ref$header === void 0 ? null : _ref$header,
      _ref$print = _ref.print,
      print = _ref$print === void 0 ? null : _ref$print,
      _ref$topFab = _ref.topFab,
      topFab = _ref$topFab === void 0 ? true : _ref$topFab;
  return /*#__PURE__*/_react["default"].createElement(_react.Fragment, null, /*#__PURE__*/_react["default"].createElement(_.NoPrint, null, topFab && /*#__PURE__*/_react["default"].createElement(_.TopFab, null), !!header && header, !!content && content), /*#__PURE__*/_react["default"].createElement(_.Print, null, !!print && print));
}

Page.propTypes = {
  topFab: _propTypes["default"].bool,
  content: _propTypes["default"].element,
  header: _propTypes["default"].element,
  print: _propTypes["default"].element
};