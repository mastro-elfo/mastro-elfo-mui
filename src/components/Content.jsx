import React from "react";

import { Box, Container } from "@material-ui/core";

/**
 * Puts a `Box` inside a `Container`.
 * @param       {[type]} children            [description]
 * @param       {Object} [ContainerProps={}] [description]
 * @param       {Object} [BoxProps={}}]      [description]
 * @constructor
 */

export default function Content({
  children,
  ContainerProps = {},
  BoxProps = {}
}) {
  return (
    <Container {...ContainerProps}>
      <Box pt={1} {...BoxProps}>
        {children}
      </Box>
    </Container>
  );
}
