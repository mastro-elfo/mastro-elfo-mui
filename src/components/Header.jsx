/**
 * Create an AppBar/Toolbar header.
 *
 * Children are rendered inside a flexGrow Typography
 */

import React, { Fragment } from "react";

import { AppBar, Toolbar } from "@material-ui/core";

import GrowTypography from "./GrowTypography";

/**
 * Create an AppBar/Toolbar header.
 *
 * Children are rendered inside a flexGrow Typography
 * @param       {[type]} children          [description]
 * @param       {[type]} [LeftAction=null] [description]
 * @param       {Array}  [RightActions=[]] [description]
 * @param       {Object} [TitleProps={}}]  [description]
 * @constructor
 */
export default function Header({
  children,
  LeftAction = null,
  RightActions = [],
  TitleProps = {}
}) {
  return (
    <Fragment>
      <AppBar>
        <Toolbar>
          {LeftAction}
          <GrowTypography variant="h6" color="inherit" {...TitleProps}>
            {children}
          </GrowTypography>
          {RightActions}
        </Toolbar>
      </AppBar>
      <Toolbar />
    </Fragment>
  );
}
