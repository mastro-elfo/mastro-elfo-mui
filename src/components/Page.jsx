/**
 * Create a single page with header, content and a "printable" version (hidden on screen).
 *
 * By default scrolling the page make a `TopFab` appear.
 */

import React, { Fragment } from "react";
import PropTypes from "prop-types";

import { NoPrint, Print, TopFab } from "./";

/**
 * Create a single page with header, content and a "printable" version (hidden on screen).
 *
 * By default scrolling the page make a `TopFab` appear.
 *
 * @param       {[type]} [content=null] [description]
 * @param       {[type]} [header=null]  [description]
 * @param       {[type]} [print=null]   [description]
 * @param       {[type]} [topFab=true}] [description]
 * @constructor
 */
export default function Page({
  content = null,
  header = null,
  print = null,
  topFab = true,
  TopFabProps = {}
}) {
  return (
    <Fragment>
      <NoPrint>
        {topFab && <TopFab {...TopFabProps} />}
        {!!header && header}
        {!!content && content}
      </NoPrint>
      <Print>{!!print && print}</Print>
    </Fragment>
  );
}

Page.propTypes = {
  content: PropTypes.element,
  header: PropTypes.element,
  print: PropTypes.element,
  topFab: PropTypes.bool,
  TopFabProps: PropTypes.object
};
