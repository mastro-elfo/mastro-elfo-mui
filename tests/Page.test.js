import React from "react";
import renderer from "react-test-renderer";

import Page from "../src/components/Page";
import Header from "../src/components/Header";
import Content from "../src/components/Content";

test("Render", () => {
  const component = renderer.create(
    <Page
      header={<Header>Header</Header>}
      content={<Content>Content</Content>}
      footer={<div>Footer</div>}
      print={<div>Print</div>}
    />
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
