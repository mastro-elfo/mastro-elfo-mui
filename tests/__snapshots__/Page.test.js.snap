// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <div
      className="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation0"
    >
      <button
        className="MuiButtonBase-root MuiFab-root makeStyles-Fab-3"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "transform": "scale(0)",
            "visibility": "hidden",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
            />
          </svg>
        </span>
      </button>
      <header
        className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h6
            className="MuiTypography-root WithStyles(ForwardRef(Typography))-root-4 MuiTypography-h6"
          >
            Header
          </h6>
        </div>
      </header>
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      />
      <div
        className="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          className="MuiBox-root MuiBox-root-5"
        >
          Content
        </div>
      </div>
      <div>
        Footer
      </div>
      <div
        style={
          Object {
            "height": "1px",
          }
        }
      />
    </div>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-6"
  >
    <div>
      Print
    </div>
  </div>,
]
`;
